browser/search_engines/search_engine_provider_service_browsertest.cc

```diff
@@ -81,31 +82,52 @@ std::string GetBraveSearchProviderSyncGUID(PrefService* prefs) {
   return data->sync_guid;
 }
 
+bool PrepopulatedDataHasDDG(PrefService* prefs) {
+  constexpr TemplateURLPrepopulateData::BravePrepopulatedEngineID
+      alt_search_providers[] = {
+          TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO,
+          TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO_DE,
+          TemplateURLPrepopulateData::
+              PREPOPULATED_ENGINE_ID_DUCKDUCKGO_AU_NZ_IE};
+
+  for (const auto& id : alt_search_providers) {
+    if (TemplateURLPrepopulateData::GetPrepopulatedEngine(prefs, id)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
 }  // namespace
 
 // Set alternative search provider prefs and check it's cleared on next
 // launching.
 IN_PROC_BROWSER_TEST_F(SearchEngineProviderServiceTest,
                        PRE_PrivateSearchProviderMigrationTest) {
-  auto* profile = browser()->profile();
-  profile->GetPrefs()->SetBoolean(
-      kShowAlternativePrivateSearchEngineProviderToggle, true);
-  profile->GetPrefs()->SetBoolean(kUseAlternativePrivateSearchEngineProvider,
-                                  true);
-  profile->GetPrefs()->ClearPref(
-      prefs::kSyncedDefaultPrivateSearchProviderGUID);
+  auto* prefs = browser()->profile()->GetPrefs();
+
+  // Running this test on region that doesn't have DDG in prepopulated data will
+  // be failed.
+  if (!PrepopulatedDataHasDDG(prefs)) {
+    return;
+  }
+
+  prefs->SetBoolean(kShowAlternativePrivateSearchEngineProviderToggle, true);
+  prefs->SetBoolean(kUseAlternativePrivateSearchEngineProvider, true);
 }
 
 IN_PROC_BROWSER_TEST_F(SearchEngineProviderServiceTest,
                        PrivateSearchProviderMigrationTest) {
-  auto* profile = browser()->profile();
-  EXPECT_FALSE(profile->GetPrefs()->GetBoolean(
-      kShowAlternativePrivateSearchEngineProviderToggle));
-  EXPECT_FALSE(profile->GetPrefs()->GetBoolean(
-      kUseAlternativePrivateSearchEngineProvider));
-  EXPECT_FALSE(profile->GetPrefs()
-                   ->GetString(prefs::kSyncedDefaultPrivateSearchProviderGUID)
-                   .empty());
+  auto* prefs = browser()->profile()->GetPrefs();
+
+  if (!PrepopulatedDataHasDDG(prefs)) {
+    return;
```

Comment:
this seems weird. Can't we set required locale in tests and validate expectations after that? Right now it looks like this test only does something good only if you're running it on a machine with a required locale. This is incorrect and will hide issues (during rebase for ex.), we shouldn't write such tests.

Reply:
You're right. Updated https://github.com/brave/brave-core/compare/8c623f4ab5cc705a31f896c5905222ab47fb39d1..3a85012adfe8ad412f32668ee5f0cf8d12b53bf5